### PR TITLE
[Artifacts] Prevent Modification of "latest" Tag

### PR DESCRIPTION
**Pull request by moranbental**
This PR ensures that users cannot modify the "latest" tag on artifact objects in the MLRun system. The "latest" tag is a critical feature for tracking the most recent version of an artifact and should only be managed by the system, not by users.
As part of this change, the following actions will be restricted:
1. `POST /projects/{project}/tags/{tag}`:  This request is used to change tags to artifacts.
2. `PUT /projects/{project}/tags/{tag}`: This request appends tags to artifacts.
3. `DELETE /projects/{project}/tags/{tag}`: This request deletes tags from artifacts. 

https://iguazio.atlassian.net/browse/ML-8684

### CODE REVIEW

File: server/py/framework/db/sqldb/db.py [hunk 1]
```diff
@@ -1390,9 +1390,11 @@ def _delete_artifacts_tags(
         commit: bool = True,
     ):
         artifacts_ids = [artifact.id for artifact in artifacts]
+        # Delete all tags except the "latest" tag, or filter by specific tags if provided
         query = session.query(ArtifactV2.Tag).filter(
             ArtifactV2.Tag.project == project,
             ArtifactV2.Tag.obj_id.in_(artifacts_ids),
+            ArtifactV2.Tag.name != "latest",
```
Comment (liranbg) [inline server/py/framework/db/sqldb/db.py]: move `latest` to constant 

PreservedTagNameLatest = "latest"

File: server/py/services/api/crud/tags.py [hunk 2]
```diff
@@ -84,7 +92,11 @@ def delete_tag_from_objects(
         delete_func = kind_to_function_names.get(tag_objects.kind, {}).get("delete")
         if not delete_func:
             raise mlrun.errors.MLRunNotFoundError(
-                f"couldn't find delete function for object kind: {tag_objects.kind}"
+                f"Couldn't find delete function for object kind: {tag_objects.kind}"
+            )
+        if tag == "latest":
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Cannot delete 'latest' tag from objects kind: {tag_objects.kind}"
```
Comment (liranbg) [inline server/py/services/api/crud/tags.py]: move out to a function (DRY)
```suggestion
        if tag == "latest":
            raise mlrun.errors.MLRunInvalidArgumentError(
                f"`latest` is a preserved tag name and cannot be deleted / mutated"
```

File: server/py/services/api/crud/tags.py [hunk 3]
```diff
@@ -46,7 +46,11 @@ def overwrite_object_tags_with_tag(
         )
         if not overwrite_func:
```
Comment (liranbg) [inline server/py/services/api/crud/tags.py:47]: id validate the tag on the endpoint before calling the CRUD to avoid thread allocation - wdyt?

**Review by liranbg — CHANGES_REQUESTED** [review]

**Review by liranbg — APPROVED** [review]

**Review by TomerShor — APPROVED** [review]
LGTM.

**Pull request merged**